boards/xtensa/m5stickc_plus/m5stickc_plus.dts

```diff
@@ -80,16 +80,65 @@
 	clock-frequency = <I2C_BITRATE_STANDARD>;
 	sda-gpios = <&gpio0 21 GPIO_OPEN_DRAIN>;
 	scl-gpios = <&gpio0 22 GPIO_OPEN_DRAIN>;
-	pinctrl-0 = <&i2c0_sda_gpio21 &i2c0_scl_gpio22>;
+	pinctrl-0 = <&i2c0_default>;
 	pinctrl-names = "default";
+
+	pmic_xpowers_axp192: pmic@34 {
+		compatible = "xpowers,axp192";
+		reg = <0x34>;
+
+		pmic_axp192_regulators: regulators {
+			compatible = "xpowers,axp192-regulator";
+
+			pmic_axp192_buck1: BUCK1 {
+				/* 700mV~3500mV@25mV/step, max output 1200mA */
+				regulator-init-microvolt = <3300000>;
+				regulator-boot-on;
```

Comment:
I am not sure if the `regulator-boot-on` option is causing the intended behavior.
```
 regulator-boot-on:
    type: boolean
    description: |
      bootloader/firmware enabled regulator.
      It's expected that this regulator was left on by the bootloader.
      If the bootloader didn't leave it on then OS should turn it on
      at boot but shouldn't prevent it from being turned off later.
      This property is intended to only be used for regulators where
      software cannot read the state of the regulator.
```
Please correct me if I am wrong, but to my knowledge the m5stack boot loader does not feature regulator support, and therefore it may start without being powered on.

Reply:
emmmm, I am not sure the bootloader on zephyr port, but i think zephyr also has some bootloader stage when boot on? 
Like this:
esp32 bootloader(1st stage) -> zephyr bootlader(2nd stage) -> zephyr OS
So this regulator-boot-on option is finish on zephyr bootlader stage?

Just my guess, i am newbie of zephyr.

Reply:
Sorry, my comment was maybe misleading. To my understanding the `regulator-boot-on` option tells the common regulator API that it may assume the regulator is already on. This could lead to the behaviour that the according regulator is not being enabled again. As the description says this option is intended to be used for regulators where software cannot read the state of the regulator.
I suppose the main reason this is still working in your case, is that you always pass "false" to the `regulator_common_init()` function, so that the common regulator api will always enable your regulator.

I doubt that any of the existing bootloaders already has support to enable the regulator.